packages/cadl-ranch-specs/http/encode/bytes/main.tsp

```diff
@@ -0,0 +1,109 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Test for encode decorator on bytes.")
+@supportedBy("dpg")
+@scenarioService("/encode/bytes")
+namespace Encode.Bytes;
+
+@encode(BytesKnownEncoding.base64)
+scalar base64Bytes extends bytes;
+
+@route("/query")
+@scenario
+@scenarioDoc("""
+Test different encode for bytes query parameters.
+Expected query parameter: 
+default=dGVzdA== (base64 encode of test)
+base64=dGVzdA==
+base64url=dGVzdA (base64url encode of test)
+base64-array=dGVzdA==, dGVzdA==
```

Comment:
Should we test base64url array, instead of base64 array (if we had to choose one)? As latter is default.